src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -362,6 +363,32 @@ public JSplitPane(int newOrientation,
 
     }
 
+    /**
+     * {@inheritDoc}
+     * @param orientation {@inheritDoc}
+     */
+    @Override
+    public void setComponentOrientation(ComponentOrientation orientation) {
+        super.setComponentOrientation(orientation);
+        Component leftComponent = this.getLeftComponent();
+        Component rightComponent = this.getRightComponent();
+        if (!this.getComponentOrientation().isLeftToRight()) {
```

Comment:
Just a suggestion, may not require to check for the `!` now as LTR or UNKNOWN can be handled together when `isLeftToRight` returns **true**.

Need a bit of code swapping between if-else block.

Reply:
yes but harmless, I guess...